src/storage/log/aof/log.rs

```diff
@@ -274,6 +290,7 @@ impl Aol {
     }
 
     pub fn close(&mut self) -> Result<()> {
+        self.check_if_fsync_failed()?;
```

Comment:
Does this mean we cannot close the Aol even though the comment on `check_if_fsync_failed` says you must close the store and restart?
[It seems like the store needs to close the aol in order to close itself.](https://github.com/surrealdb/surrealkv/blob/aca3678480fcd4329b009da584cee43da3a21a03/src/storage/kv/store.rs#L445-L446)

Can we perhaps say you cannot gracefully close the store(along with aol) when there was a fsync failure, and instead you need to forcefully restart the store somehow, and we need to implement a way to restart the store forcefully?

Reply:
Thank you for having a detailed look @mumoshu !

You raise a valid point, I think for close this check is not needed as you have rightly mentioned because we need to fsync to update the metadata of the directory during close. I am trying to repair on restart as you've reviewed in this [draft PR](https://github.com/surrealdb/surrealkv/pull/35). I'll try to remove this check.